### ISSUE TITLE
NVDA doesn't read MathML in IE for this case

### ISSUE THREAD

**Issue by NSoiffer** [nvaccess/nvda#6527]
NVDA 2016.3 doesn't seem to recognize the MathML in IE 11 in this particular example produced by MathJax. Some other MathJax examples seem to work fine, although I think I have run into this problem before. The page also works in Firefox, so this seems to be IE specific, but I don't know why this case differs from some others.

The example is at https://s.codepen.io/nms/debug/ZpgBer.

I have removed all the JavaScript (i.e., MathJax), so hopefully it is easier to debug/fix. Instructions are on the web page.

For reference, the overall structure is
```
...
<h2>
<div>
   <span class="MathJax" ...>
      <nobr aria-hidden="true">
        ugly spans to position the math for display
     </nobr>

     <span class="MJX_Assistive_MathML" ...>
        <math xmlns="http://www.w3.org/1998/Math/MathML"> .... </math>
    </span>
  </span>
</div>   
...
```
CSS makes the MathML in invisible.

**Comment by jcsteh** [comment]
The math does seem to be picked up here. However, I do see two lines of "x y" before the proper math. That is, the real math is on the third line after the heading. Can you confirm this?

Obviously, the extraneous content before the math is annoying, but if the math can be accessed beneath it, that lowers the priority somewhat.

**Comment by NSoiffer** [comment]
In IE 11, I never hear the math. One down arrow gives "x y" and another down arrow gives "Heading level 2: Notes". In Firefox, it reads the math appropriately after that first arrow.

In the HTML, there is a aria-hidden="true" on a <nobr> tag and that is followed by the MathML, so there should only be one thing being read.

It sounds like you are getting a different behavior than I am (hearing both "x y" and "x over y") in IE. Does that happen with a single down arrow or do you down arrow a second time to get the "x over y" speech? Do you get the proper behavior in Firefox, so you can compare the difference between what should be happening and what is happening?

To make sure we have the same version of IE, I have IE update version 11.0.36 (Version 11.0.0900.18499). As mentioned earlier, I have NVDA V2016.3.

**Comment by jcsteh** [comment]
Yes, I see the math after 2 presses of down arrow in IE. I see the correct behaviour (no extraneous lines) in Firefox.

I'm aware that the aria-hidden *should* hide the raw content, but IE has some really stupid bugs relating to inclusion of hidden content in name calculation. The question is how we'll hack around this one...

What version of Windows are you testing with? I'm using Windows 10. Also, do you have the MathPlayer IE plugin enabled or disabled?

**Comment by NSoiffer** [comment]
I originally tried it using Windows 7. I've tried it with the MathPlayer plugin (both MathPlayer and its BHO) enable and disabled and don't see a difference.

I retried it on a Windows 10 machine with IE11, same thing. I tried it with the MathPlayer plugin enabled and disabled there also.

Although NVDA doesn't support Edge, I tried that too. The behavior was different -- it took three down arrows before I heard "x y".

In all those case, I never heard "x over y".

I wonder what the difference between your machine and my machine is?

**Comment by bhavyashah** [comment]
@jcsteh Since you seem to have been involved in diagnosing and troubleshooting this issue, could you please highlight the current standing and impact of this issue as well as assign an appropriate priority to this ticket?

**Comment by jcsteh** [comment]
I was never able to reproduce the issue as reported; i.e. being unable to read the "x over y". I did see the extraneous content, but that's somewhat lower priority. It'd be good to know if anyone else can reproduce this. STR:

1. Ensure you have MathPlayer installed.
2. Open Internet Explorer and go to https://codepen.io/nms/full/ZpgBer
3. Press 2 to jump to the heading level 2.
4. Press down arrow twice. (This is necessary to skip the extraneous content noted above.)
5. Press down arrow again.
6. Confirm whether or not you hear "x over y".

**Comment by bhavyashah** [comment]
On Internet Explorer 11, the third line below the first heading at
level 2 does read as x over y, although two lines preceding that read
only as x y. Unless @NSoiffer states that he/she still experiences
this bug, I suggest closing as worksforme.

On 8/7/17, James Teh <notifications@github.com> wrote:
> I was never able to reproduce the issue as reported; i.e. being unable to
> read the "x over y". I did see the extraneous content, but that's somewhat
> lower priority. It'd be good to know if anyone else can reproduce this.
> STR:
>
> 1. Ensure you have MathPlayer installed.
> 2. Open Internet Explorer and go to https://codepen.io/nms/full/ZpgBer
> 3. Press 2 to jump to the heading level 2.
> 4. Press down arrow twice. (This is necessary to skip the extraneous content
> noted above.)
> 5. Press down arrow again.
> 6. Confirm whether or not you hear "x over y".
>
> --
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/6527#issuecomment-320543391


-- 
Best Regards
Bhavya Shah

Blogger at Hiking Across Horizons: https://bhavyashah125.wordpress.com/

Contacting Me
E-mail Address: bhavya.shah125@gmail.com
Follow me on Twitter @BhavyaShah125 or www.twitter.com/BhavyaShah125
Mobile Number: +91 7506221750

**Comment by Adriani90** [comment]
I am closing this as works for me. As @bhavyashah already stated, the third line after the heading 2 is the math ml object which you can enter with nvda+alt+m anc everything can be read acordingly. The third line is reported also as "x over y".
